scripts/projects/projects.ts

```diff
@@ -0,0 +1,102 @@
+import fs from 'fs';
+import {Project} from "../../src/types";
+import { dump, load } from 'js-yaml';
+import { fileURLToPath } from 'url';
+import { dirname } from 'path';
+import { Importer } from "./Importer";
+import { GithubImporter } from "./GithubImporter";
+import { DoraHacksImporter } from "./DoraHacksImporter";
+import { EthGlobalImporter } from "./EthGlobalImporter";
+
+const registeredImporters: Map<RegExp, Importer> = new Map();
+registeredImporters.set(/github\.com/i, new GithubImporter());
+registeredImporters.set(/dorahacks\.io/i, new DoraHacksImporter());
+registeredImporters.set(/ethglobal\.com/i, new EthGlobalImporter());
+
+const [,, command, ...args] = process.argv;
+
+function loadProjects(): Map<string, Project> {
+    try {
+        const projectsPath = dirname(fileURLToPath(import.meta.url))+'/../../projects';
+        return new Map(
+            fs.readdirSync(projectsPath)
+                .filter(dir => dir !== '.template')
+                .map(dir => [
+                    dir,
+                    load(fs.readFileSync(`${projectsPath}/${dir}/description.yaml`, 'utf8')) as Project
+                ])
+        );
+    } catch (error) {
+        console.error('Error loading projects:', error);
+        return new Map();
+    }
+}
+
+switch (command) {
+    case 'add':
+        if (args.length < 1) {
+            console.error('Usage: add <url> [directory]');
+            process.exit(1);
+        }
+        const [url, directory] = args;
+
+        let matched=false;
+        for (const [regex, importer] of registeredImporters) {
+            if (url.match(regex)) {
```

Comment:
SInce only one importer is expected to match then I would flatten the code
```ts
const match = registeredImporters.find(([regex, importer]) => url.match(regex))
if (!match) {
    ...
    process.exit(1);
}
const importer = match[1]
...
```